ambry-store/src/main/java/com.github.ambry.store/Log.java

```diff
@@ -319,13 +319,9 @@ private void initialize(List<LogSegment> segmentsToLoad, long segmentCapacityInB
       segmentsToLoad = Collections.singletonList(checkArgsAndGetFirstSegment(segmentCapacityInBytes));
     }
 
-    long totalSegments = -1;
+    LogSegment firstSegment = segmentsToLoad.get(0);
+    long totalSegments = firstSegment.getName().isEmpty() ? 1 : capacityInBytes / firstSegment.getCapacityInBytes();
```

Comment:
do you validate that total capacity is a multiple of segment capacities anywhere during initialization?